### ISSUE TITLE
Jump on enter from error/warning in interactive mode

### ISSUE THREAD

**Issue by qrilka** [haskell/haskell-mode#1015]
I remember this feature was there in haskell-mode some time ago (maybe even more than year or two) and it was quite handy but currently it does not work.
My haskell-mode config currently is quite basic:
```
(add-hook 'haskell-mode-hook 'turn-on-haskell-indentation)
(require 'haskell-interactive-mode)
(require 'haskell-process)
(add-hook 'haskell-mode-hook 'interactive-haskell-mode)
(custom-set-variables
 '(haskell-process-type 'cabal-repl)
 '(haskell-process-suggest-remove-import-lines t)
 '(haskell-process-auto-import-loaded-modules t)
 '(haskell-process-log t))
```
I've updated to the most recent version from ELPA and run Emacs 24.5

**Comment by bergey** [comment]
On my computer, it works, like this:

 `RET` is bound to `haskell-interactive-mode-return`, which calls `haskell-interactive-next-error-function`.

If I'm on an error message like 
```
src/Diagrams/Angle.hs:38:18: Could not find module ‘Control.Applicativ’ …
    Perhaps you meant Control.Applicative (from base-4.8.1.0)
    Use -v to see a list of the files searched for.
```

`RET` jumps to the file / line / char specified in the error message.

@qrilka can you verify that `RET` is bound to `haskell-interactive-mode-return` on your emacs, and paste an example of an error message that doesn't work?

**Comment by gracjan** [comment]
Related #1170.

**Comment by gracjan** [comment]
And it seems that it overall works, @qrilka: write here and we will reopen the issue if it still is a problem.

**Comment by qrilka** [comment]
No, it does not work for me, `RET` is bound to `haskell-interactive-mode-return` but pressing it with cursor on the error message does not do anything and nothing appears in `*Messages*`
I use the latest 13.18 version from melpa.
Is there any other way I could debug this problem? it annoys a bit :(

**Comment by gracjan** [comment]
Reopening in this case. @qrilka, can you help debugging it?

**Comment by qrilka** [comment]
I'd be glad to help though I'm not an advanced emacser

**Comment by alios** [comment]
I had the very same problem: Pressing RET on an Error Line simply does nothing any more.

I had a ```(global-flycheck-mode 1)``` in my ```.emacs```. Flycheck seems to somewhat hook in and causing this problem.

After commenting the line above out (or do ```M-x flycheck-mode``` in a running session), the jumping to source works again. 

Seems as if flycheck somewhat hooks in there.

**Comment by qrilka** [comment]
oh thanks @alios for the info - `M-x flycheck-mode` did the trick for the currently used config from https://bitbucket.org/k_bx/dotfiles/src today

**Comment by qrilka** [comment]
@gracjan does it mean that haskell-mode isn't supposed to work with `global-flycheck-mode`?

**Comment by gracjan** [comment]
As I understood this discussion the issue is in flycheck, isn't it?

**Comment by fice-t** [comment]
@gracjan More so that Flycheck is hijacking `next-error-function` while the jumping depends on that variable being `haskell-interactive-next-error-function`.

It would be nice if the current behaviour could be made to work with Flycheck.

**Comment by gracjan** [comment]
Lets leave this open in this case.
